shell/platform/embedder/embedder.h

```diff
@@ -861,6 +861,64 @@ typedef struct {
   const char* variant_code;
 } FlutterLocale;
 
+typedef int64_t FlutterEngineDartPort;
+
+typedef enum {
+  kFlutterEngineDartObjectTypeNull,
+  kFlutterEngineDartObjectTypeBool,
+  kFlutterEngineDartObjectTypeInt32,
+  kFlutterEngineDartObjectTypeInt64,
+  kFlutterEngineDartObjectTypeDouble,
+  kFlutterEngineDartObjectTypeString,
+  kFlutterEngineDartObjectTypeBuffer,
+} FlutterEngineDartObjectType;
+
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterEngineDartBuffer).
+  size_t struct_size;
+  /// An opaque baton passed back to the embedder when the
+  /// buffer_collect_callback is invoked. The engine does not interpret this
+  /// field in any way.
+  void* user_data;
+  /// This is an optional field.
+  ///
+  /// When specified, the engine will assume that the buffer is owned by the
+  /// embedder. When the data is no longer needed by any isolate, this callback
+  /// will be made on an internal engine managed thread. The embedder is free to
+  /// collect the buffer here. When this field is specified, it is the embedders
+  /// responsibility to keep the buffer alive and unmodified till this callback
+  /// is invoked by the engine. The user data specified in the callback is the
+  /// value of `user_data` field in this struct.
+  ///
+  /// When NOT specified, the VM creates an internal copy of the buffer. The
+  /// caller is free to modify the buffer as necessary or collect it immediately
+  /// after the call to `FlutterEnginePostDartObject`.
+  VoidCallback buffer_collect_callback;
+  /// A pointer to the bytes of the buffer.
+  const uint8_t* buffer;
```

Comment:
Consider removing 'const'. Dart code will be able to write to this buffer in the external case.

Reply:
Oh. I actually did not know that. I updated the const-ness and the docs to mention that. I also added a note to indicate the the page protections not restrict writing to the buffer.

Reply:
@rmacnak-google: I noticed the string value is not const either. Does that mean that the string is not copied out by the VM either?